featuretools/primitives/install.py

```diff
@@ -0,0 +1,181 @@
+
+    # if it isn't local, download it. if remote, it must be archive
+    if not (os.path.isdir(directory_or_archive) or os.path.isfile(directory_or_archive)):
+        directory_or_archive = download_archive(directory_or_archive)
+
+    # if archive, extract directory to temp folders
+    if os.path.isfile(directory_or_archive):
+        directory = extract_archive(directory_or_archive)
+    else:
+        directory = directory_or_archive
+
+    # Iterate over all the files and determine the primitives to install
+    files = list_primitive_files(directory)
+    all_primitives = {}
+    files_to_copy = []
+    for filepath in files:
+        primitive_name, primitive_obj = load_primitive_from_file(filepath)
+        files_to_copy.append(filepath)
+        all_primitives[primitive_name] = primitive_obj
+
+    # before installing, confirm with user
+    primitives_list = ", ".join(all_primitives.keys())
+    if prompt:
+        while True:
+            resp = input("Install primitives: %s? (Y/n) " % primitives_list)
+            if resp == "Y":
+                break
+            elif resp == "n":
+                return
+    else:
+        print("Installing primitives: %s" % primitives_list)
+
+    # copy the files
+    installation_dir = get_installation_dir()
+    for to_copy in tqdm(files_to_copy):
+        shutil.copy2(to_copy, installation_dir)
+
+    # clean up tmp dir
+    if os.path.exists(tmp_dir):
+        shutil.rmtree(tmp_dir)
+
+
+def get_featuretools_root():
+    return os.path.dirname(featuretools.__file__)
+
+
+def get_installation_dir():
+    "return the path to the installation directory with in featuretools"
+    installation_dir = os.path.join(get_featuretools_root(), "primitives", "installed")
+    return installation_dir
+
+
+def get_installation_temp_dir():
+    """Returns the path to the installation directory with in featuretools.
+
+        If the directory, doesn't exist it is created
+    """
+    tmp_dir = os.path.join(get_installation_dir(), ".tmp/")
+    if not os.path.exists(tmp_dir):
+        os.makedirs(tmp_dir)
+    return os.path.join(get_installation_dir(), ".tmp/")
+
+
+def download_archive(uri):
+    # determine where to save locally
+    filename = os.path.basename(urlparse(uri).path)
+    local_archive = os.path.join(get_installation_temp_dir(), filename)
+
+    with open(local_archive, 'wb') as f:
+        try:
+            remote_archive = smart_open(uri, 'rb', ignore_extension=True)
+        except NoCredentialsError:
+            # fallback to anonymous using s3fs
+            s3 = s3fs.S3FileSystem(anon=True)
+            remote_archive = s3.open(uri, 'rb')
+
+        f.write(remote_archive.read())
+
+        remote_archive.close()
+
+    return local_archive
+
+
+def extract_archive(filepath):
+    if (filepath.endswith("tar.gz")):
+        tar = tarfile.open(filepath, mode='r:gz')
+    elif (filepath.endswith("tar")):
+        tar = tarfile.open(filepath, "r:")
```

Comment:
we should probably handle the non-tar case so there's a user-friendly error

Reply:
done